pointers/pointers.c

```diff
@@ -53,7 +68,23 @@ void string_copy(char *x, char *y)
 */
 int string_compare(char *m, char *n)
 {
-
+  while (*m != '\0') {
+      if (*n == '\0') {
+          return 1;
+      }
+      if (*m > *n) {
+          return 1;
+      }
+      if (*n > *m) {
+          return -1;
+      }
+      m++;
+      n++;
+  }
+  if(*n != '\0') {
+      return -1;
+  }
+  return 0;
```

Comment:
This function can be streamlined by just returning the difference between `m` and `n`.